### PR TITLE
feat: social kyc ui

### PR DESCRIPTION
**Pull request by ice-hades**

### CODE REVIEW

File: src/api/kyc/verifySocialKYCStep.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import {post} from '@api/client';
+import {SocialKycStepNumber} from '@api/tokenomics/types';
+import {SupportedLocale} from '@translations/localeConfig';
+
+export type SocialMethod = 'facebook' | 'twitter';
+
+type SocialKycData = {
+  expectedPostText: string;
+  remainingAttempts: number;
+  result: 'SUCCESS';
```
Comment (ice-ares) [inline src/api/kyc/verifySocialKYCStep.ts]: Or failure

File: src/screens/SocialKycFlow/InstructionsStep/index.tsx [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    Vibration.vibrate([0, 50]);
+  };
+  const socialKycStatus = useSelector(socialKycStatusSelector);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    if (socialKycMethod === 'Facebook') {
+      if (isSocialKycFinalized(socialKycStatus)) {
+        updateStepPassed();
+      } else if (socialKycStatus === 'ERROR') {
+        onSkip();
```
Comment (ice-ares) [inline src/screens/SocialKycFlow/InstructionsStep/index.tsx:83]: Dont. You only skip when user presses on “not now”

File: src/screens/SocialKycFlow/InstructionsStep/index.tsx [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+  const repostText = useSelector(socialKycRepostTextSelector);
+  const copyText = () => {
+    Clipboard.setString(repostText);
+    setIsTextCopied(true);
+    Vibration.vibrate([0, 50]);
+  };
+  const socialKycStatus = useSelector(socialKycStatusSelector);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    if (socialKycMethod === 'Facebook') {
```
Comment (ice-ares) [inline src/screens/SocialKycFlow/InstructionsStep/index.tsx:79]: I dont see the twitter logic here

File: src/screens/SocialKycFlow/ResultStep/index.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+};
+
+export function ResultStep({onTryAgain, onSkip}: Props) {
+  const socialKycStatus = useSelector(socialKycStatusSelector);
+  const socialKycAttempts = useSelector(socialKycAttemptsSelector);
+  const canContinue =
+    socialKycStatus === 'SUCCESS' ||
+    (socialKycStatus === 'FAILED' && !socialKycAttempts);
+  const onContinue = () => {
+    onSkip();
```
Comment (ice-ares) [inline src/screens/SocialKycFlow/ResultStep/index.tsx:38]: don't skip.

File: src/screens/SocialKycFlow/SelectProfileStep/components/SocialKycMethodSelectionTile/index.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  socialKycMethod: SocialKycMethod;
+};
+
+export function SocialKycMethodSelectionTile({
+  setSocialKycMethod,
+  selectedSocialKycMethod,
+  socialKycMethod,
+}: Props) {
+  const isSelected = selectedSocialKycMethod === socialKycMethod;
+  console.log({isSelected});
```
Comment (ice-ares) [inline src/screens/SocialKycFlow/SelectProfileStep/components/SocialKycMethodSelectionTile/index.tsx]: remove it

File: src/screens/SocialKycFlow/SelectProfileStep/index.tsx [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    }
+  };
+  const getSocialKycRepostTextStatus = useSelector(
+    getSocialKycRepostTextStatusSelector,
+  );
+  useEffect(() => {
+    if (getSocialKycRepostTextStatus === 'SUCCESS') {
+      updateStepPassed();
+    } else if (getSocialKycRepostTextStatus === 'ERROR') {
+      onSkip();
```
Comment (ice-ares) [inline src/screens/SocialKycFlow/SelectProfileStep/index.tsx:57]: dont skip

File: src/screens/SocialKycFlow/VerificationStep/index.tsx [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+  onSkip,
+}: Props) {
+  const [repostLinkUrl, setRepostLinkUrl] = useState('');
+  const socialKycStatus = useSelector(socialKycStatusSelector);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    if (isSocialKycFinalized(socialKycStatus)) {
+      updateStepPassed();
+    } else if (socialKycStatus === 'ERROR') {
+      onSkip();
```
Comment (ice-ares) [inline src/screens/SocialKycFlow/VerificationStep/index.tsx:74]: dont skip

File: src/store/modules/Tokenomics/sagas/startMiningSession.ts [hunk 8]
```diff
@@ -109,6 +109,14 @@ export function* startMiningSessionSaga(
           });
           return;
         }
+        if (errorData.kycSteps.includes(3) || errorData.kycSteps.includes(5)) {
```
Comment (ice-ares) [inline src/store/modules/Tokenomics/sagas/startMiningSession.ts]: use those specific types u created

File: src/screens/SocialKycFlow/InstructionsStep/index.tsx (hunk 2 above)
Comment (ice-hades) [reply src/screens/SocialKycFlow/InstructionsStep/index.tsx:83]: When step is not passed as an argument it just start the mining

File: src/screens/SocialKycFlow/InstructionsStep/index.tsx (hunk 3 above)
Comment (ice-hades) [reply src/screens/SocialKycFlow/InstructionsStep/index.tsx:79]: special case for facebook only

File: src/screens/SocialKycFlow/ResultStep/index.tsx (hunk 4 above)
Comment (ice-hades) [reply src/screens/SocialKycFlow/ResultStep/index.tsx:38]: When step is not passed as an argument it just start the mining

File: src/screens/SocialKycFlow/SelectProfileStep/index.tsx (hunk 6 above)
Comment (ice-hades) [reply src/screens/SocialKycFlow/SelectProfileStep/index.tsx:57]: When step is not passed as an argument it just start the mining

File: src/screens/SocialKycFlow/VerificationStep/index.tsx (hunk 7 above)
Comment (ice-hades) [reply src/screens/SocialKycFlow/VerificationStep/index.tsx:74]: When step is not passed as an argument it just start the mining

File: src/store/modules/Tokenomics/sagas/startMiningSession.ts (hunk 8 above)
Comment (ice-hades) [reply src/store/modules/Tokenomics/sagas/startMiningSession.ts]: those are types not constants. can't be used here

Comment (ice-hades) [reply src/store/modules/Tokenomics/sagas/startMiningSession.ts]: added constants

File: src/screens/SocialKycFlow/SelectProfileStep/components/SocialKycMethodSelectionTile/index.tsx (hunk 5 above)
Comment (ice-hades) [reply src/screens/SocialKycFlow/SelectProfileStep/components/SocialKycMethodSelectionTile/index.tsx]: removed

File: src/api/kyc/verifySocialKYCStep.ts (hunk 1 above)
Comment (ice-hades) [reply src/api/kyc/verifySocialKYCStep.ts]: added

**Review by ice-ares — DISMISSED** [review]

File: src/store/modules/SocialKyc/sagas/getSocialKycRepostText.ts [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+      yield put(
+        SocialKycActions.GET_SOCIAL_KYC_REPOST_TEXT.SUCCESS.create({
+          repostText: result.expectedPostText,
+        }),
+      );
+    } else {
+      yield put(SocialKycActions.GET_SOCIAL_KYC_REPOST_TEXT.ERROR.create());
+    }
+  } catch (error: unknown) {
+    yield put(SocialKycActions.GET_SOCIAL_KYC_REPOST_TEXT.ERROR.create());
```
Comment (ice-ares) [inline src/store/modules/SocialKyc/sagas/getSocialKycRepostText.ts:38]: do u need this even if u ignore it and start mining? (for the case bellow)

Comment (ice-hades) [reply src/store/modules/SocialKyc/sagas/getSocialKycRepostText.ts:38]: this line starts mining. line below shows the monkey for all errors except 2 specific

**Review by ice-ares — DISMISSED** [review]

File: src/screens/SocialKycFlow/InstructionsStep/index.tsx [hunk 10]
```diff
@@ -0,0 +1,231 @@
...
+  onGoBack,
+  onSkip,
+  kycStep,
+}: Props) {
+  const [isTextCopied, setIsTextCopied] = useState(false);
+  const repostText = useSelector(socialKycRepostTextSelector);
+  const copyText = () => {
+    Clipboard.setString(repostText);
+    setIsTextCopied(true);
+    Vibration.vibrate([0, 50]);
```
Comment (ice-hector) [inline src/screens/SocialKycFlow/InstructionsStep/index.tsx]: For start mining we use `hapticFeedback()` function from `src/utils/device.ts`. Maybe we should reuse it here or its not the same? I remember we had some bugs related to vibration work/not work

File: src/screens/SocialKycFlow/InstructionsStep/index.tsx [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+      if (isSocialKycFinalized(socialKycStatus)) {
+        updateStepPassed();
+      } else if (socialKycStatus === 'ERROR') {
+        onSkip();
+      }
+    }
+  }, [kycStep, onSkip, socialKycMethod, socialKycStatus, updateStepPassed]);
+  const onConfirm = () => {
+    if (socialKycMethod === 'Facebook') {
+      getFacebookAccessTokenForUserPosts()
```
Comment (ice-hector) [inline src/screens/SocialKycFlow/InstructionsStep/index.tsx:90]: I don't think it's a good place to handle this token behavior here, UI should know nothing about this, pls move to the separate saga

File: src/screens/SocialKycFlow/VerificationStep/hooks/useCountdown.ts [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+  }, []);
+
+  useEffect(() => {
+    stopCountdown();
+  }, [stopCountdown]);
+
+  const startCountdown = useCallback(() => {
+    countdownIntervalRef.current = setInterval(
+      () =>
+        setCountdown(c => {
```
Comment (ice-hector) [inline src/screens/SocialKycFlow/VerificationStep/hooks/useCountdown.ts]: Please don't use one letter variable naming

File: src/screens/SocialKycFlow/components/StepInstruction/index.tsx [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+    backgroundColor: COLORS.primaryLight,
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  stepNumberText: {
+    ...font(15, null, 'black', 'white', 'center'),
+  },
+  text: {
+    flex: 1,
+    paddingLeft: isRTL && isAndroid ? null : rem(12),
```
Comment (ice-hector) [inline src/screens/SocialKycFlow/components/StepInstruction/index.tsx]: I see this same logic for paddings in many places

```
    paddingLeft: isRTL && isAndroid ? null : rem(12),
    paddingRight: isRTL && isAndroid ? rem(12) : null,
```

please create one style for this case, export it, and use it when needed (so we can change that padding in one place if needed)

File: src/screens/SocialKycFlow/InstructionsStep/index.tsx (hunk 11 above)
Comment (ice-hades) [reply src/screens/SocialKycFlow/InstructionsStep/index.tsx:90]: ok. but facebook flow is disabled for now. so no way to check those changes. so will leave it as is until facebook is enabled

Comment (ice-hector) [reply src/screens/SocialKycFlow/InstructionsStep/index.tsx:90]: Ok then add some TODO here so we not forget about it when we enable the facebook flow

File: src/screens/SocialKycFlow/InstructionsStep/index.tsx (hunk 10 above)
Comment (ice-hades) [reply src/screens/SocialKycFlow/InstructionsStep/index.tsx]: fixed

File: src/screens/SocialKycFlow/InstructionsStep/index.tsx (hunk 11 above)
Comment (ice-hades) [reply src/screens/SocialKycFlow/InstructionsStep/index.tsx:90]: fixed

File: src/screens/SocialKycFlow/VerificationStep/hooks/useCountdown.ts (hunk 12 above)
Comment (ice-hades) [reply src/screens/SocialKycFlow/VerificationStep/hooks/useCountdown.ts]: fixed

File: src/screens/SocialKycFlow/components/StepInstruction/index.tsx (hunk 13 above)
Comment (ice-hades) [reply src/screens/SocialKycFlow/components/StepInstruction/index.tsx]: fixeed

**Review by ice-hector — APPROVED** [review]

**Pull request merged**